File: notes/bubble-sort-example/bubble-sort.ts
function bubbleSort(arr: number[]) {
  let isSwapped = true;
  while (isSwapped) {
    isSwapped = false;

    for (let i = 0; i < arr.length; i++) {
      let current = arr[i];
      let next = arr[i + 1];

      if (next > current) {
        arr[i + 1] = current;
        arr[i] = next;
        isSwapped = true;
      }
    }
  }

  return arr;
}

function main() {
  console.log("apples are red");
  const mySort = bubbleSort([3, 1, 5, 6, 9]);
  console.log(mySort);
}

File: notes/bubble-sort-example/bubble-sort.rs
fn main() {
    let sort_me = vec![1, 3, 6, 2, 9, 4];
    let sorted_arr = bubble_sort(sort_me);
    println!("{:?}", sorted_arr);
}

fn bubble_sort(mut arr: Vec<u8>) -> Vec<u8> {
    let mut is_swapped = true;

    while is_swapped {
        is_swapped = false;
        let length = arr.len();

        for i in 0..length - 1 {
            let current = arr[i];
            let next = arr[i + 1];

            if current > next {
                arr[i] = next;
                arr[i + 1] = current;
                is_swapped = true;
            }
        }
    }
    return arr;
}

// the o of n for this woudl be O(n *  n) cause firslt there are 2 loops
// which are nested, so the operation happens twice
